src/hdf5_init.c

```diff
@@ -0,0 +1,177 @@
+#include "hdf5_init.h"
+#include "blosc_filter.h"
+
+#include <blosc.h>
+#include <hdf5.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+/*
+// functions for debugging
+void check_space_and_storage(hid_t dset) {
+        H5D_space_status_t space_status;
+        hsize_t storage_size;
+
+        status = H5Dget_space_status(dset, &space_status);
+        storage_size = H5Dget_storage_size(dset);
+        printf("Space for dataset has%sbeen allocated. \n", space_status ==
+H5D_SPACE_STATUS_ALLOCATED ? " " : " NOT "); printf("Storage size for dataset is
+: %ld bytes.\n", (long)storage_size);
+}
+*/
+
+// REAL functions
+void initialize_file_and_plist(char *filename,
+                               hid_t *file_id,
+                               hid_t *fapl_id,
+                               hid_t *fcpl_id)
+{
+  unsigned mode = H5F_ACC_TRUNC;
+  char *version, *date;
+
+  if ((*fcpl_id = H5Pcreate(H5P_FILE_CREATE)) == H5I_INVALID_HID) {
+    fprintf(stderr, "Error in creating fcpl in create_file\n");
+    return;
+  }
+  if ((*fapl_id = H5Pcreate(H5P_FILE_ACCESS)) == H5I_INVALID_HID) {
+    fprintf(stderr, "Error in creating fapl in create_file\n");
+    return;
+  } else {
+    if (H5Pset_alignment(*fapl_id, 1024, 4096) < 0) {
+      fprintf(stderr, "Error in H5Pset_alignment\n");
+      return;
+    }
+  }
+  if ((*file_id = H5Fcreate(filename, mode, *fcpl_id, *fapl_id)) ==
+      H5I_INVALID_HID) {
+    fprintf(stderr, "Error in creating file_id in create_file, please check if "
+                    "file already existed\n");
+    return;
+  }
+  if (register_blosc(&version, &date) < 0) {
+    fprintf(stderr, "Error in register_blosc\n");
+    return;
+  } else {
+    printf("Blosc version info: %s (%s)\n", version, date);
+    free(version);
+    free(date);
+  }
+}
+
+void initialize_lcpl(hid_t *lcpl_id)
```

Comment:
Here `lcpl` is initialised separately, I suggest either initialise all plists separately like this, or bundle them with the above function.